### DISCUSSION TITLE
Trouble migrating to 1.113.1

### DISCUSSION THREAD

**Discussion by pnuthamper** [immich-app/immich#12364] [Q&A] [answered]
i jumped a few releases (sorry i didnt capture the version number before hand but it was probably latest in december 2023).

i upgrade to the same version of the android client, then ran 'docker compose pull && docker compose up -d'

i didnt see any issues, but i noticed i had a number of errors related to thumbnails etc.

so i figured i better make sure my docker-compose and .env are updated too...

i see there are some new env variables (such as DB_DATA_LOCATION) and some changes to docker-compose.yml
i went ahead and copied over docker-compose.yml and made the handful of edits (external storage for example).

now when i try to run docker compose up i get this error:

```
WARN[0000] The "DB_DATA_LOCATION" variable is not set. Defaulting to a blank string. 
invalid spec: :/var/lib/postgresql/data: empty section between colons
```

i hadnt previously customized anything in var/lib/postgres

any pointers on what to look at next? thanks!

**Comment by alextran1502** [comment]
You must have updated your `docker-compose.yml` file. Please edit the file so that it has the `pgdata` volume as shown below

<img width="1502" alt="image" src="https://github.com/user-attachments/assets/479cd554-9e13-4a58-ad7f-821ab6fc9c2a">

**Reply by pnuthamper** [reply]
@alextran1502 thanks for your prompt reply. this got me thinking i should revert my docker-compose.yml and debug the original post-upgrade error. perhaps i was premature in changing that file. This is the original post-upgrade error:

```
immich_server            | [Nest] 17  - 09/05/2024, 5:42:06 PM   ERROR [Api:LoggerRepository~hk73mlma] Unable to send file: Error
immich_server            | Error: ENOENT: no such file or directory, access 'upload/profile/532a4da9-fddd-4cb8-9275-7e0150db9768/6f385c1a-1ed7-4cb7-9fc0-9b50711a112e.jpg'
immich_server            |     at async access (node:internal/fs/promises:606:10)
immich_server            |     at async sendFile (/usr/src/app/dist/utils/file.js:55:9)
immich_server            |     at async UserController.getProfileImage (/usr/src/app/dist/controllers/user.controller.js:71:9)
immich_server            | [Nest] 17  - 09/05/2024, 5:42:06 PM    WARN [Api:ExpressAdapter~hk73mlma] Content-Type doesn't match Reply body, you might need a custom ExceptionFilter for non-JSON responses
immich_server            | [Nest] 17  - 09/05/2024, 5:42:06 PM   ERROR [Api:ExceptionsHandler~hk73mlma] ENOENT: no such file or directory, access 'upload/profile/532a4da9-fddd-4cb8-9275-7e0150db9768/6f385c1a-1ed7-4cb7-9fc0-9b50711a112e.jpg'
```

**Reply by pnuthamper** [reply]
i get many of these in server logs and images wont load in the web client

**Reply by alextran1502** [reply, marked as answer]
yeah, probably reverting the docker-compose file, then reading through the release notes to update would be the best route.

If you have less than 5000 assets in total, starting from scratch wouldn't be a bad idea

**Reply by pnuthamper** [reply]
thanks Alex. unfortunately I have about 10,000 assets so I'd like to save that as a last resort. Looking through the release notes from my last version (which i figured out was april 2024 time frame, not certain which, 1.101-x.103.x) I didn't find any breaking changes that I needed to adopt. Looking in logs I see a large number of broken paths with this root:

/usr/src/app/upload/thumbs/<some guid>

so it seems a thumbnail issue goes with the error trace i listed above.
Is there any easy way to recover this by blowing away thumbnails and just rebulding those?

**Reply by pnuthamper** [reply]
> thanks Alex. unfortunately I have about 10,000 assets so I'd like to save that as a last resort. Looking through the release notes from my last version (which i figured out was april 2024 time frame, not certain which, 1.101-x.103.x) I didn't find any breaking changes that I needed to adopt. Looking in logs I see a large number of broken paths with this root:
> 
> /usr/src/app/upload/thumbs/
> 
> so it seems a thumbnail issue goes with the error trace i listed above. Is there any easy way to recover this by blowing away thumbnails and just rebulding those?

actually its not just thumbs, i see some `ERROR [Microservices:JobService] Error: ENOENT: no such file or directory, stat` errors for the paths `upload/upload/` and `upload/profile`.

Seems somehow there's some file path issues but i have reverted my docker-compose.yml changes so hopefully i can recover this?

i dont mind rerunning all the jobs to build out metadata and thumbs but it would be nice not to start from scratch. my backup is from april but thats pretty old given how many assets I create per month.

**Reply by pnuthamper** [reply]
ended up just setting the paths in .env explicitly to point to the existing directories that were in use. thanks for the replies and for maintaining this great project
